hermeto/core/package_managers/maven/main.py

```diff
@@ -0,0 +1,397 @@
+DEFAULT_LOCKFILE_NAME = "lockfile.json"
+
+
+class MavenComponentInfo(TypedDict):
+    """Contains the data needed to generate a maven SBOM component."""
+
+    name: str
+    purl: str
+    version: str
+    scope: str
+    included: bool
+    missing_hash_in_file: Optional[Path]
+
+
+class ResolvedMavenPackage(TypedDict):
+    """Contains all of the data for a resolved maven package."""
+
+    package: MavenComponentInfo
+    dependencies: list[MavenComponentInfo]
+    projectfiles: list[ProjectFile]
+
+
+class MavenDependency:
+    """A Maven dependency from lockfile.json."""
+
+    def __init__(self, dependency_dict: dict[str, Any]) -> None:
+        """Initialize a MavenDependency.
+
+        :param dependency_dict: the raw dict for a dependency from lockfile.json
+        """
+        self._dependency_dict = dependency_dict
+
+    @property
+    def group_id(self) -> str:
+        """Get the group ID."""
+        return self._dependency_dict["groupId"]
+
+    @property
+    def artifact_id(self) -> str:
+        """Get the artifact ID."""
+        return self._dependency_dict["artifactId"]
+
+    @property
+    def version(self) -> str:
+        """Get the version."""
+        return self._dependency_dict["version"]
+
+    @property
+    def name(self) -> str:
+        """Get the full name (groupId:artifactId)."""
+        return f"{self.group_id}:{self.artifact_id}"
+
+    @property
+    def scope(self) -> str:
+        """Get the dependency scope."""
+        return self._dependency_dict.get("scope", "compile")
+
+    @property
+    def checksum(self) -> Optional[str]:
+        """Get the checksum."""
+        return self._dependency_dict.get("checksum")
+
+    @property
+    def checksum_algorithm(self) -> Optional[str]:
+        """Get the checksum algorithm."""
+        return self._dependency_dict.get("checksumAlgorithm")
+
+    @property
+    def resolved_url(self) -> Optional[str]:
+        """Get the resolved URL."""
+        return self._dependency_dict.get("resolved")
+
+    @property
+    def included(self) -> bool:
+        """Return True if this dependency should be included."""
+        return self._dependency_dict.get("included", True)
+
+    @property
+    def children(self) -> list[dict[str, Any]]:
+        """Get the children dependencies."""
+        return self._dependency_dict.get("children", [])
+
+    def to_component_info(self) -> MavenComponentInfo:
+        """Convert to MavenComponentInfo."""
+        purl = PackageURL(
+            type="maven",
+            namespace=self.group_id,
+            name=self.artifact_id,
+            version=self.version,
+        )
+
+        return MavenComponentInfo(
+            name=self.name,
+            purl=str(purl),
+            version=self.version,
+            scope=self.scope,
+            included=self.included,
+            missing_hash_in_file=None if self.checksum else Path("lockfile.json"),
```

Comment:
`Path("lockfile.json")` will be relative to the current working directory, is this expected? _Edit_: `Path` is used as a refined `str` in this context, this probably should be mentioned somewhere. Or are there any plans to use it as `Path` in the future?